src/v/wasm/ffi.h

```diff
@@ -266,7 +274,7 @@ std::tuple<Type> extract_parameter(
         return std::make_tuple(static_cast<Type>(nullptr));
     } else if constexpr (std::is_pointer_v<Type>) {
         // Assume this is an out val
-        auto guest_ptr = static_cast<uint32_t>(raw_params[idx++]);
+        auto guest_ptr = ptr(static_cast<uint32_t>(raw_params[idx++]));
```

Comment:
micro-nit: `ptr guest_ptr {static_cast<uint32_t>(raw_params[idx++])};`

Reply:
I'm used to the advice recommended in https://abseil.io/tips/88 (sorry former googler here), but I've still updated this.